[Upgrade Assistant] use request hook

alisonelizabeth opened the pull request:
This PR introduces the use of the `useRequest`/`sendRequest` helpers from `es_ui_shared` in Upgrade Assistant. As we add more functionality to UA, it will be helpful to leverage these to simplify client requests.

### Notable changes
- All HTTP requests were updated on the client, **except** the reindexing requests. The logic for this is a little more complex, and since we have plans to make changes to the reindexing UX, I think it makes sense to address any refactoring at that time. 
- `tabs.tsx` was converted from a class to function component in order to use the request hook.
- I've deleted `tabs.test.tsx` and `checkup_tab.test.tsx` in favor of CITs.

### How to test
Hopefully we have solid test coverage to catch any issues. It would be helpful to do a quick smoke test of the UI though and verify the client requests (toggling deprecation logging, updating deprecated index settings, etc.). It's a bit hacky to test on master. You can leverage this commit to pull in the hacks: https://github.com/elastic/kibana/commit/613bddf2914153d7c77fa626e671a2b54fa0d5aa. In addition, you will need to create the indices specified in the mocked response:

```
PUT test1
{
  "settings": {
    "index" : {
      "refresh_interval": "1m"
      }      
  }
}

PUT test2
{
  "settings": {
    "index" : {
      "refresh_interval": "1m"
      }      
  }
}
```

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/)
* Commit: d3f89463f9065f93ee0f96beb3ef589f4b034db4
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Quick Commit Checks](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/execution/node/431/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/execution/node/109/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/","number":"112832","commit":"d3f89463f9065f93ee0f96beb3ef589f4b034db4"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/)
* Commit: 013fd56a867dc596785aab17a8c6f19347a05c7b
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/execution/node/163/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/execution/node/460/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/execution/node/122/log)

### History
* :broken_heart: [Build #112832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/) failed d3f89463f9065f93ee0f96beb3ef589f4b034db4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/","number":"112832","commit":"d3f89463f9065f93ee0f96beb3ef589f4b034db4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/","number":"112919","commit":"013fd56a867dc596785aab17a8c6f19347a05c7b"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/)
* Commit: 246936d1a0ca95479dcfd5505c72ee3d83f057a9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/246936d1a0ca95479dcfd5505c72ee3d83f057a9)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/a4fa4a62cb7d518b4911f56b8ef84e445b351c57) | [after](https://github.com/elastic/kibana/commit/246936d1a0ca95479dcfd5505c72ee3d83f057a9) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 122 | 124 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a4fa4a62cb7d518b4911f56b8ef84e445b351c57) | [after](https://github.com/elastic/kibana/commit/246936d1a0ca95479dcfd5505c72ee3d83f057a9) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 136.4KB | 135.9KB | -515.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/a4fa4a62cb7d518b4911f56b8ef84e445b351c57) | [after](https://github.com/elastic/kibana/commit/246936d1a0ca95479dcfd5505c72ee3d83f057a9) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 33.4KB | 33.7KB | +237.0B |

### History
* :broken_heart: [Build #112919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/) failed 013fd56a867dc596785aab17a8c6f19347a05c7b
* :broken_heart: [Build #112832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/) failed d3f89463f9065f93ee0f96beb3ef589f4b034db4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/","number":"112832","commit":"d3f89463f9065f93ee0f96beb3ef589f4b034db4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/","number":"112919","commit":"013fd56a867dc596785aab17a8c6f19347a05c7b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/","number":"113082","commit":"246936d1a0ca95479dcfd5505c72ee3d83f057a9"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/)
* Commit: d6b734c84b425dccfc33e5a311a856a8dc1918ca
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/d6b734c84b425dccfc33e5a311a856a8dc1918ca)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f14ac90e7d05e83321bf571ebec62c0213ecd59e) | [after](https://github.com/elastic/kibana/commit/d6b734c84b425dccfc33e5a311a856a8dc1918ca) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 126 | 128 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f14ac90e7d05e83321bf571ebec62c0213ecd59e) | [after](https://github.com/elastic/kibana/commit/d6b734c84b425dccfc33e5a311a856a8dc1918ca) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 144.1KB | 143.6KB | -512.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f14ac90e7d05e83321bf571ebec62c0213ecd59e) | [after](https://github.com/elastic/kibana/commit/d6b734c84b425dccfc33e5a311a856a8dc1918ca) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 33.1KB | 33.3KB | +237.0B |

### History
* :green_heart: [Build #113082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/) succeeded 246936d1a0ca95479dcfd5505c72ee3d83f057a9
* :broken_heart: [Build #112919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/) failed 013fd56a867dc596785aab17a8c6f19347a05c7b
* :broken_heart: [Build #112832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/) failed d3f89463f9065f93ee0f96beb3ef589f4b034db4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/","number":"112832","commit":"d3f89463f9065f93ee0f96beb3ef589f4b034db4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/","number":"112919","commit":"013fd56a867dc596785aab17a8c6f19347a05c7b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/","number":"113082","commit":"246936d1a0ca95479dcfd5505c72ee3d83f057a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/","number":"113268","commit":"d6b734c84b425dccfc33e5a311a856a8dc1918ca"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/)
* Commit: f810fbf908bf9431a8dd60d640f567201c6ade75
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/f810fbf908bf9431a8dd60d640f567201c6ade75)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/4d27af99c49229146e066aadb59dfbfb73443066) | [after](https://github.com/elastic/kibana/commit/f810fbf908bf9431a8dd60d640f567201c6ade75) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 126 | 128 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/4d27af99c49229146e066aadb59dfbfb73443066) | [after](https://github.com/elastic/kibana/commit/f810fbf908bf9431a8dd60d640f567201c6ade75) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 144.1KB | 143.8KB | -353.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/4d27af99c49229146e066aadb59dfbfb73443066) | [after](https://github.com/elastic/kibana/commit/f810fbf908bf9431a8dd60d640f567201c6ade75) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 33.1KB | 33.3KB | +237.0B |

### History
* :green_heart: [Build #113268](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/) succeeded d6b734c84b425dccfc33e5a311a856a8dc1918ca
* :green_heart: [Build #113082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/) succeeded 246936d1a0ca95479dcfd5505c72ee3d83f057a9
* :broken_heart: [Build #112919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/) failed 013fd56a867dc596785aab17a8c6f19347a05c7b
* :broken_heart: [Build #112832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/) failed d3f89463f9065f93ee0f96beb3ef589f4b034db4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/","number":"112832","commit":"d3f89463f9065f93ee0f96beb3ef589f4b034db4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/","number":"112919","commit":"013fd56a867dc596785aab17a8c6f19347a05c7b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/","number":"113082","commit":"246936d1a0ca95479dcfd5505c72ee3d83f057a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/","number":"113268","commit":"d6b734c84b425dccfc33e5a311a856a8dc1918ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/","number":"113558","commit":"f810fbf908bf9431a8dd60d640f567201c6ade75"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/)
* Commit: b6316fdb4504a02f90e177952b5a5e3dae7cc93f
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/b6316fdb4504a02f90e177952b5a5e3dae7cc93f)
* This comment will update when the build is complete

### Failed CI Steps
* [fetch latest from 'master' at origin](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/execution/node/142/log)
* [Execute xpack-kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/execution/node/749/log)
* [Execute xpack-kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/execution/node/943/log)
* [fetch latest from 'master' at origin](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/execution/node/85/log)

### History
* :green_heart: [Build #113558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/) succeeded f810fbf908bf9431a8dd60d640f567201c6ade75
* :green_heart: [Build #113268](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/) succeeded d6b734c84b425dccfc33e5a311a856a8dc1918ca
* :green_heart: [Build #113082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/) succeeded 246936d1a0ca95479dcfd5505c72ee3d83f057a9
* :broken_heart: [Build #112919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/) failed 013fd56a867dc596785aab17a8c6f19347a05c7b
* :broken_heart: [Build #112832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/) failed d3f89463f9065f93ee0f96beb3ef589f4b034db4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112832/","number":"112832","commit":"d3f89463f9065f93ee0f96beb3ef589f4b034db4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/","number":"112919","commit":"013fd56a867dc596785aab17a8c6f19347a05c7b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/","number":"113082","commit":"246936d1a0ca95479dcfd5505c72ee3d83f057a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/","number":"113268","commit":"d6b734c84b425dccfc33e5a311a856a8dc1918ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/","number":"113558","commit":"f810fbf908bf9431a8dd60d640f567201c6ade75"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/","number":"113717","commit":"b6316fdb4504a02f90e177952b5a5e3dae7cc93f"}]}
PIPELINE-->

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113808/)
* Commit: a018fd5da9237a6d0a070bd4eb1987ee00637c8b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/a018fd5da9237a6d0a070bd4eb1987ee00637c8b)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/645545ebdf0d98aa35e9aa143028185c0ec90dc1) | [after](https://github.com/elastic/kibana/commit/a018fd5da9237a6d0a070bd4eb1987ee00637c8b) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 126 | 128 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/645545ebdf0d98aa35e9aa143028185c0ec90dc1) | [after](https://github.com/elastic/kibana/commit/a018fd5da9237a6d0a070bd4eb1987ee00637c8b) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 144.1KB | 143.8KB | -342.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/645545ebdf0d98aa35e9aa143028185c0ec90dc1) | [after](https://github.com/elastic/kibana/commit/a018fd5da9237a6d0a070bd4eb1987ee00637c8b) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 33.1KB | 33.3KB | +237.0B |

### History
* :broken_heart: [Build #113717](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/) failed b6316fdb4504a02f90e177952b5a5e3dae7cc93f
* :green_heart: [Build #113558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/) succeeded f810fbf908bf9431a8dd60d640f567201c6ade75
* :green_heart: [Build #113268](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/) succeeded d6b734c84b425dccfc33e5a311a856a8dc1918ca
* :green_heart: [Build #113082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/) succeeded 246936d1a0ca95479dcfd5505c72ee3d83f057a9
* :broken_heart: [Build #112919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/) failed 013fd56a867dc596785aab17a8c6f19347a05c7b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112919/","number":"112919","commit":"013fd56a867dc596785aab17a8c6f19347a05c7b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/","number":"113082","commit":"246936d1a0ca95479dcfd5505c72ee3d83f057a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/","number":"113268","commit":"d6b734c84b425dccfc33e5a311a856a8dc1918ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/","number":"113558","commit":"f810fbf908bf9431a8dd60d640f567201c6ade75"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/","number":"113717","commit":"b6316fdb4504a02f90e177952b5a5e3dae7cc93f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113808/","number":"113808","commit":"a018fd5da9237a6d0a070bd4eb1987ee00637c8b"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

jloleysens reviewed (approved):
Thanks for working on this @alisonelizabeth ! The code looks in really good shape.

I left a few non-blocker comments in the code.

While testing this out locally all of the network requests worked 100% fine. While using the hacks + manually created indices to test out the reindexing flow I ran into the following while reindexing `test2`:

<img width="1187" alt="Screenshot 2021-03-18 at 15 10 46" src="https://user-images.githubusercontent.com/8155004/111640359-6ae78c00-87fc-11eb-8738-f9179629135c.png">

Not sure this is relevant to these changes so I don't think its worth blocking on. Just not sure what lead to this error state so sharing FYI. Perhaps you have seen this?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113957/)
* Commit: b1586d337fb8629abefba509a6b06c446b46bdb5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/b1586d337fb8629abefba509a6b06c446b46bdb5)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/628bb4b3775aba5fee822e76fc8152ccc6866087) | [after](https://github.com/elastic/kibana/commit/b1586d337fb8629abefba509a6b06c446b46bdb5) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 126 | 128 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/628bb4b3775aba5fee822e76fc8152ccc6866087) | [after](https://github.com/elastic/kibana/commit/b1586d337fb8629abefba509a6b06c446b46bdb5) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 144.1KB | 143.8KB | -342.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/628bb4b3775aba5fee822e76fc8152ccc6866087) | [after](https://github.com/elastic/kibana/commit/b1586d337fb8629abefba509a6b06c446b46bdb5) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 33.1KB | 33.3KB | +237.0B |

### History
* :green_heart: [Build #113808](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113808/) succeeded a018fd5da9237a6d0a070bd4eb1987ee00637c8b
* :broken_heart: [Build #113717](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/) failed b6316fdb4504a02f90e177952b5a5e3dae7cc93f
* :green_heart: [Build #113558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/) succeeded f810fbf908bf9431a8dd60d640f567201c6ade75
* :green_heart: [Build #113268](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/) succeeded d6b734c84b425dccfc33e5a311a856a8dc1918ca
* :green_heart: [Build #113082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/) succeeded 246936d1a0ca95479dcfd5505c72ee3d83f057a9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113082/","number":"113082","commit":"246936d1a0ca95479dcfd5505c72ee3d83f057a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/","number":"113268","commit":"d6b734c84b425dccfc33e5a311a856a8dc1918ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/","number":"113558","commit":"f810fbf908bf9431a8dd60d640f567201c6ade75"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/","number":"113717","commit":"b6316fdb4504a02f90e177952b5a5e3dae7cc93f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113808/","number":"113808","commit":"a018fd5da9237a6d0a070bd4eb1987ee00637c8b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113957/","number":"113957","commit":"b1586d337fb8629abefba509a6b06c446b46bdb5"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/components/tabs.tsx:
```diff
@@ -228,26 +82,104 @@ export class UpgradeAssistantTabs extends React.Component<Props, TabsState> {
-    this.setState({ telemetryState: TelemetryState.Running });
+  useEffect(() => {
+    if (isLoading === false) {
+      setTelemetryState(TelemetryState.Running);
```
Good catch. That was a mistake on my end - fixed now.

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/tests_client_integration/indices.test.ts:
```diff
@@ -15,79 +15,169 @@ describe('Indices tab', () => {
+          find(`${accordionTestSubj}.removeIndexSettingsButton`).simulate('click');
+        });
+
+        const modal = document.body.querySelector(
+          '[data-test-subj="indexSettingsDeleteConfirmModal"]'
```
The modal is not inside the component DOM tree, so we need to access it via `document.body`. We're following a similar pattern in our other CITs. I've added a comment to the test so it's more clear.

alisonelizabeth commented:
Thanks for the review @jloleysens! I've addressed your feedback.

> While testing this out locally all of the network requests worked 100% fine. While using the hacks + manually created indices to test out the reindexing flow I ran into the following while reindexing test2:

That's strange. I just ran through the flow myself again and it worked as expected. This error would occur if there is an error fetching the index settings. I don't think it's related to this PR as I did not make changes to the reindexing flow. I'll keep an eye on it though.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114021/)
* Commit: 5973c33380289ff3a68d4e6a09ef9f0890197c43
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94473/5973c33380289ff3a68d4e6a09ef9f0890197c43)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/628bb4b3775aba5fee822e76fc8152ccc6866087) | [after](https://github.com/elastic/kibana/commit/5973c33380289ff3a68d4e6a09ef9f0890197c43) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 126 | 128 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/628bb4b3775aba5fee822e76fc8152ccc6866087) | [after](https://github.com/elastic/kibana/commit/5973c33380289ff3a68d4e6a09ef9f0890197c43) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 144.1KB | 143.8KB | -342.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/628bb4b3775aba5fee822e76fc8152ccc6866087) | [after](https://github.com/elastic/kibana/commit/5973c33380289ff3a68d4e6a09ef9f0890197c43) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 33.1KB | 33.3KB | +237.0B |

### History
* :green_heart: [Build #113957](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113957/) succeeded b1586d337fb8629abefba509a6b06c446b46bdb5
* :green_heart: [Build #113808](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113808/) succeeded a018fd5da9237a6d0a070bd4eb1987ee00637c8b
* :broken_heart: [Build #113717](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/) failed b6316fdb4504a02f90e177952b5a5e3dae7cc93f
* :green_heart: [Build #113558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/) succeeded f810fbf908bf9431a8dd60d640f567201c6ade75
* :green_heart: [Build #113268](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/) succeeded d6b734c84b425dccfc33e5a311a856a8dc1918ca

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113268/","number":"113268","commit":"d6b734c84b425dccfc33e5a311a856a8dc1918ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113558/","number":"113558","commit":"f810fbf908bf9431a8dd60d640f567201c6ade75"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113717/","number":"113717","commit":"b6316fdb4504a02f90e177952b5a5e3dae7cc93f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113808/","number":"113808","commit":"a018fd5da9237a6d0a070bd4eb1987ee00637c8b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113957/","number":"113957","commit":"b1586d337fb8629abefba509a6b06c446b46bdb5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114021/","number":"114021","commit":"5973c33380289ff3a68d4e6a09ef9f0890197c43"}]}
PIPELINE-->
